src/impl/interaction/library/Pan.ts

```diff
@@ -12,186 +12,106 @@
+        }
 
-    private configMove(touched: StdState, cancelled: CancellingState, moved: StdState, dataHandler?: PanFSMDataHandler): void {
-        const isGuardMoveKO = (evt: TouchEvent): boolean => evt.changedTouches[0].identifier === this.touchID &&
-            !this.isStable(evt.changedTouches[0].clientX, evt.changedTouches[0].clientY);
-
-        const moveTouched = new TouchMoveTransition(touched, cancelled);
-        moveTouched.isGuardOK = isGuardMoveKO;
-
-        const moveCancelled = new TouchMoveTransition(moved, cancelled);
-        moveCancelled.isGuardOK = isGuardMoveKO;
-
-        const isGuardMoveOK = (evt: TouchEvent): boolean => evt.changedTouches[0].identifier === this.touchID &&
-            this.isStable(evt.changedTouches[0].clientX, evt.changedTouches[0].clientY);
-        const actionMoveOK = (event: TouchEvent): void => {
-            dataHandler?.panning(event);
-        };
-
-        const moveTouchedOK = new TouchMoveTransition(touched, moved);
-        moveTouchedOK.isGuardOK = isGuardMoveOK;
-        moveTouchedOK.action = actionMoveOK;
-
-        const moveMovedOK = new TouchMoveTransition(moved, moved);
-        moveMovedOK.isGuardOK = isGuardMoveOK;
-        moveMovedOK.action = actionMoveOK;
-    }
-
-
-    private configRelease(moved: StdState, cancelled: CancellingState, released: TerminalState, dataHandler?: PanFSMDataHandler): void {
-        const releaseMoved = new TouchReleaseTransition(moved, cancelled);
-        releaseMoved.isGuardOK = (evt: TouchEvent): boolean => evt.changedTouches[0].identifier === this.touchID &&
-            !this.checkFinalPanConditions(evt);
-
-        const releaseFinal = new TouchReleaseTransition(moved, released);
-        releaseFinal.isGuardOK = (evt: TouchEvent): boolean => evt.changedTouches[0].identifier === this.touchID &&
-            this.checkFinalPanConditions(evt);
-        releaseFinal.action = (event: TouchEvent): void => {
-            dataHandler?.panned(event);
-        };
-    }
-
-    protected setInitialValueOnTouch(evt: TouchEvent): void {
-        const touch: Touch = evt.changedTouches[0];
-        this.touchID = touch.identifier;
-        this.moveAxe = this.horizontal ? touch.clientX : touch.clientY;
-        this.stableAxe = this.horizontal ? touch.clientY : touch.clientX;
-    }
+        const touches = this.getConccurFSMs()
+            .filter(fsm => fsm.getTouchId() === event.changedTouches[0].identifier);
 
-    protected checkFinalPanConditions(evt: TouchEvent): boolean {
-        return this.getPanDistance(evt.changedTouches[0].clientX, evt.changedTouches[0].clientY) >= this.minLength;
-    }
+        if (touches.length > 0) {
+            return touches[0].process(event);
+        }
 
-    public override reinit(): void {
-        super.reinit();
-        this.touchID = undefined;
-        this.stableAxe = undefined;
-        this.moveAxe = undefined;
+        return this.getConccurFSMs().some(conccurFSM => conccurFSM.process(event));
```

Comment:
Why  in reinit processing an event on one concurr FSM?

Reply:
This is in process() actually, I replaced the entire Pan code with the MultiTouch() code so the diff is confusing